lib/launch-direct.c

```diff
@@ -927,13 +932,15 @@ make_appliance_dev (guestfs_h *g, int virtio_scsi)
 
   /* Calculate the index of the drive. */
   ITER_DRIVES (g, i, drv) {
-    if (virtio_scsi) {
-      if (drv->iface == NULL || STREQ (drv->iface, "ide"))
-        index++;
-    }
-    else /* virtio-blk */ {
-      if (drv->iface == NULL || STRNEQ (drv->iface, "virtio"))
-        index++;
+    if (drv->device == device_disk) {
```

Comment:
Couldn't CDROM be IDE/Virtio?

Reply:
here is a function to calculate last index of drives which is ext2 appliance. CDROM drives have "/dev/srX" names.